wallet-new/integration/Functions.hs

```diff
@@ -110,26 +136,23 @@ runActionCheck walletClient walletState actionProb = do
         log $ "Skipped actions: " <> show ([minBound..maxBound] \\ nub succs)
 
 
--- | Attempt each action in the list. If an action fails, ignore the
--- failure and try the next action in the sequence.
-tryAll :: Alternative f => NonEmpty (f a) -> f a
-tryAll (x :| xs) = foldl' (\this next -> (this *> next) <|> next) x xs
-
-freshPassword :: (MonadState WalletState m, MonadIO m) => m SpendingPassword
+freshPassword :: (MonadState WalletState (PropertyM m), MonadIO m) => PropertyM m SpendingPassword
 freshPassword = do
     passwords <- gets (toList . view walletsPass)
-    liftIO $ generate $ arbitrary `suchThat` (`notElem` passwords)
+    pick $ arbitrary `suchThat` (`notElem` passwords)
 
 -- | Here we run the actions.
 {-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
 -- NOTE: ordNub is available in latest universum, but current universum-0.9 doesn't have it.
 {-# ANN module ("HLint: ignore Use ordNub" :: Text) #-}
 runAction
-    :: (WalletTestMode m, MonadState WalletState m)
+    :: (WalletTestMode m,
+        MonadState WalletState m,
+        MonadState WalletState (PropertyM m),
+        MonadThrow (PropertyM m))
```

Comment:
These constraints should not be necessary -- you will only need one `MonadState` constraint, and the `MonadThrow` constraint is unnecessary.